server/scripts/feed-data.ts

```diff
@@ -1,139 +1,182 @@
+  { fileName: 'cohort_participants.csv', table: collections.COHORT_PARTICIPANTS },
+  { fileName: 'participants_status.csv', table: collections.PARTICIPANTS_STATUS },
+  { fileName: 'participant_post_hire_status.csv', table: collections.PARTICIPANT_POST_HIRE_STATUS },
+  { fileName: 'return_of_service_status.csv', table: collections.ROS_STATUS },
+];
+
+/**
+ * Adds the `created_by` and `updated_by` fields to a row object. This is required for database insertion.
+ * @param row Row of data to append the fields to
+ * @param userID Value to populate `created_by` and `updated_by` with, defaults to `'system'`
+ * @returns Input row with the system fields populated
+ */
+function addSystemFields(row: CsvRow, userID = 'system') {
+  return { ...row, created_by: userID, updated_by: userID };
+}
+
+/**
+ * Inserts a single row worth of data into a given table in the database.
+ * @param row Data for a single row to insert into the database
+ * @param table Name of table to insert the row into
+ * @returns Result of the operation
+ */
+async function insertRow(row: CsvRow, table: string): Promise<InsertResult> {
+  // Type guard for DB - generally unneeded, but satisfies strict TS stanards.
+  if (!('db' in dbClient) || !dbClient.db) throw new Error('Database failed to initialize!');
+  try {
+    await dbClient.db[table].insert(addSystemFields(row));
+    return { id: row.id, table, status: InsertStatus.SUCCESS };
+  } catch (error) {
+    // Type guard for unexpected error types (i.e. non-DB errors)
+    if (!error || typeof error !== 'object' || !('code' in error) || !('detail' in error)) {
+      logWithLevel('Unexpected error type during insertion!', 'error');
+      throw error;
+    }
+    // WARN: This also catches duplicate errors from related tables, such as `participants_distance` if populating `employer_sites`.
+    // Ideally that should have its own case, but generally that ends up causing FK errors that get caught later anyways.
+    if (error.code === '23505') return { id: row.id, table, status: InsertStatus.DUPLICATE };
+    if (error.code === '23503') {
+      logWithLevel(`${error.detail} Row ${row.id} \x1b[4mwill not be inserted\x1b[24m.`, 'warn');
+      return { id: row.id, table, status: InsertStatus.MISSING_FK };
+    }
+    throw error;
+  }
+}
+
+/**
+ * Inserts all rows from a streamed CSV file into a table in the database.
+ * @param filePath Name of file to read from
+ * @param table Name of table in DB to write to
+ * @returns Array of `InsertResult`s for each transaction
+ */
+function insertCSV(filePath: string, table: string) {
+  return new Promise<InsertResult[]>((resolve, reject) => {
+    const rowResults: Promise<InsertResult>[] = [];
+    parseFile(filePath, { headers: true })
+      .on('error', reject)
+      .on('data', (row) => rowResults.push(insertRow(row, table)))
+      .on('end', (rowCount: number) => {
+        dbClient.runRawQuery(`ALTER SEQUENCE ${table}_id_seq RESTART WITH ${rowCount}`);
```

Comment:
Add a comment explaining why this is needed - it's not a bad change, but not immediately clear.